enclave/src/lib.rs

```diff
@@ -0,0 +1,247 @@
+
+#![crate_name = "sampleenclave"]
+#![crate_type = "staticlib"]
+
+#![cfg_attr(not(target_env = "sgx"), no_std)]
+#![cfg_attr(target_env = "sgx", feature(rustc_private))]
+
+#[cfg(not(target_env = "sgx"))]
+#[macro_use]
+extern crate sgx_tstd as std;
+extern crate sgx_rand;
+extern crate sgx_rand_derive;
+extern crate sgx_serialize;
+
+#[macro_use]
+extern crate sgx_serialize_derive;
+
+extern crate sgx_types;
+extern crate sgx_tseal;
+extern crate sgx_tcrypto;
+extern crate sgx_trts;
+extern crate sgx_tse;
+
+extern crate rustls;
+extern crate webpki;
+extern crate webpki_roots;
+extern crate itertools;
+extern crate base64;
+extern crate httparse;
+extern crate yasna;
+extern crate bit_vec;
+extern crate num_bigint;
+extern crate chrono;
+
+use std::sgxfs::SgxFile;
+use std::io::{Read, Write};
+
+use sgx_serialize::{SerializeHelper, DeSerializeHelper};
+
+use std::prelude::v1::*;
+use std::time::*;
+
+use sgx_types::*;
+use sgx_tcrypto::*;
+use std::time::SystemTime;
+use std::untrusted::time::SystemTimeEx;
+use std::str;
+use std::vec::Vec;
+use std::convert::TryInto;
+use std::string::String;
+
+use num_bigint::BigUint;
+use bit_vec::BitVec;
+use yasna::models::ObjectIdentifier;
+use chrono::Duration;
+use chrono::TimeZone;
+use chrono::Utc as TzUtc;
+
+use sgx_tse::rsgx_create_report;
+
+const CERTEXPIRYDAYS: i64 = 90i64;
+const ISSUER : &str = "confidential-token-broker";
+const SUBJECT : &str = "confidential-token-broker";
+
+#[derive(Serializable, DeSerializable, Clone, Default, Debug)]
+struct RSAKeyData {
+    modulus: Vec<u8>,
+    d: Vec<u8>,
+    e: Vec<u8>,
+    not_befer: time_t,
+    not_after: time_t,
+}
+
+// const RSA_DURATION: u64 = 604800;
+const RSA_DURATION: u64 = 330;
+const ADVANCE_REFRESH_TIME: i64 = 300;
+
+static mut RSA_KEY_DATA: Option<RSAKeyData> = None;
+
+// extern "C" {
+//     pub fn ocall_get_target_info(
+//         ret_val  : *mut sgx_status_t,
+//         ti       : *mut sgx_target_info_t
+//     ) -> sgx_status_t;
+
+//     pub fn ocall_generate_quote(
+//         ret_val        : *mut sgx_status_t,
+//         p_report       : *mut sgx_report_t,
+//         p_quote        : *mut u8,
+//         max_quote_len  : u32,
+//         p_quote_len    : *mut u32
+//     ) -> sgx_status_t;
+// }
+
+
+fn write_rsa_key(path: &str) -> sgx_status_t {
```

Comment:
If this function is not public, we may not need to use `sgx_status_t` as the return type, you can directly use standard rust return type `Result<...>` and then you can use `?` micro to handle the error  gracefully.

Reply:
Okay, during testing, all the functions were called as ecall functions. I didn't notice this when making modifications. I will make sure to correct it later. Thank you for your guidance, teacher.